aips/aip-4.md

```diff
@@ -0,0 +1,78 @@
+# AIP-4: Multi-Composable Token Standard
+
+### This standard allows the coexistence of soulbound and non-soulbound NFTs.
+
+### Summary
+
+This standard is an extension of [Aptos-Token](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-token/sources/token.move). It proposes an interface allowing each token to define its soulbound property directly.
+
+### Motivation
+
+Soulbound tokens have enormous potential in a wide variety of applications. However, with no standard, soulbound tokens are incompatible with the Aptos-Token standard. Consensus on a common standard is required to allow for broader ecosystem adoption, as well as to make it easier for application developers to build expressive applications on top of the standard.
+
+This AIP (Aptos Improvement Proposal) envisions soulbound tokens as non-transferable NFTs that will unlock primitives such as robust proof of attendance, credentials, distributed governance rights, and much more. However, soulbound tokens must be distinguishable from [Aptos-Tokens](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-token/sources/token.move) to provide this utility.
+
+### Rationale
+
+Extending this utility to the existing Aptos-Token standard provides significant benefits:
+
+- Consensus on a soulbound token standard
+- Monitoring a single event store for soulbound and non-soulbound tokens
+- Compatibility with the Aptos-Token standard.
+
+An alternative solution to the non-transferability problem is creating a "Locking Standard", where tokens can be deposited, claimed, and burned. Creating a locker that acts as a wrapper around tokens essentially makes them soulbound. This solution has a significant drawback: monitoring can no longer be handled for both soulbound and non-soulbound tokens in a single event store, thus creating the need for separate monitoring of an entirely different event store. This is undesirable from usability, composability, and scalability standpoints.
+
+### Specification
+
+Modifications required to the current [Aptos-Token](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-token/sources/token.move) implementation:
+
+- Append “soulbound: bool” too *Tokens* and *Collections.*
+- Minting of soulbound tokens can only be handled through *“claims”*. After a soulbound NFT has been claimed it is locked to the receiver with the only valid operation being *“burn”,* if applicable to the token’s configuration*.*
+- Validate before any transfer execution to ensure the token isn’t soulbound.
+
+```rust
+/// Executed when a specific token_id is set or updated to soulbound, according to the parameter "bounded".
```

Comment:
does this mean a token can mutate between a soulbound and non-soulbound token? In my mind, soulbound should not be transferrable after minting to its owner

Reply:
No, the soulbound property should never be mutatable. The validation checks the existing transfer functions to ensure that the current token is not soulbound. This validation is needed as soulbound and non-soulbound tokens can coexist.